R/reporting_triangle-methods.R

```diff
@@ -0,0 +1,495 @@
+#' @title Subset reporting_triangle objects
+#'
+#' @description
+#' Extract or replace parts of a reporting_triangle object while preserving its
+#' attributes.
+#'
+#' @param x A [reporting_triangle] object
+#' @param ... Indices for subsetting
+#' @return A [reporting_triangle] object with the subset data
+#' @family reporting_triangle
+#' @method [ reporting_triangle
+#' @export
+`[.reporting_triangle` <- function(x, ...) {
+  # Store attributes before subsetting
+  old_class <- class(x)
+  old_delays_unit <- get_delays_unit(x)
+
+  out <- NextMethod()
+
+  # If result is not a matrix (e.g., single row/column as vector),
+  # return as-is
+  if (!is.matrix(out)) {
+    return(out)
+  }
+
+  # Restore class and attributes
+  class(out) <- old_class
+  attr(out, "delays_unit") <- old_delays_unit
+
+  # Validate the subsetted result
+  assert_reporting_triangle(out)
+
+  return(out)
+}
+
+#' @title Subset assignment for reporting_triangle objects
+#'
+#' @description
+#' Assignment method that allows modification of reporting_triangle values
+#' while validating the result remains a valid reporting_triangle.
+#'
+#' @param x A [reporting_triangle] object.
+#' @param ... Row and column indices for assignment.
+#' @param value Values to assign.
+#' @return The modified reporting_triangle object.
+#' @family reporting_triangle
+#' @export
+#' @method [<- reporting_triangle
+`[<-.reporting_triangle` <- function(x, ..., value) {
+  # Perform the assignment using NextMethod
+  x_modified <- NextMethod()
+
+  # Preserve class and attributes if still a matrix
+  if (is.matrix(x_modified)) {
+    class(x_modified) <- class(x)
+    attr(x_modified, "delays_unit") <- get_delays_unit(x)
```

Comment:
As written, only the delays unit is actually stored as an attribute correct? Because reference dates are now row names, and max delay and structure can be detected from the reporting triangle